### DISCUSSION TITLE
Cant build eclipse repository

### DISCUSSION THREAD

**Discussion by jdebsucks** [eclipse-tycho/tycho#1279] [Q&A]
I am having trouble getting a maven tycho "eclipse repository" to build with Java 17 and maven 3.6.3
![mvn-tycho build](https://user-images.githubusercontent.com/111005295/186018475-7f1061f1-82c8-453a-b273-5a6b69a24fe9.PNG)
All of the eclipse features/plugins build fine but the "eclipse repository" cannot the feature defined in my product file. I am more than happy to add additional files to assist with troubleshooting! 
Parent pom
`
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
	<modelVersion>4.0.0</modelVersion>
	<packaging>pom</packaging>

	<parent>
		<my normal parent pom, setup for java 17/>
	</parent>

	<groupId>cse.rssm.gui</groupId>
	<artifactId>cse.rssm.gui.parent</artifactId>
	<version>3.13.0-SNAPSHOT</version>

	<properties>
		<tycho.version>2.7.4</tycho.version>
		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
		<maven.test.jvmargs />
		<version>3.13.0-SNAPSHOT</version>
	</properties>

	<modules>
		<module>../cse.rssm.gui.utility.m2osgi</module>
		<module>../org.jfree.experimental.chart.swt</module>
		<module>../cse.rssm.gui.utility.swt</module>
		<module>../cse.rssm.gui.utility.model</module>
		<module>../cse.rssm.gui.utility.properties</module>
		<module>../cse.rssm.gui.utility.preferences</module>
		<module>../cse.rssm.gui.utility.validators</module>
		<module>../cse.rssm.gui.utility.security</module>
		<module>../cse.rssm.gui.utility.elements</module>
		<module>../cse.rssm.gui.utility.elements.tests</module>
		<module>../cse.rssm.gui.utility.alarming</module>
		<module>../cse.rssm.gui.wizards.rfsourceinput</module>
		<module>../cse.rssm.gui.wizards.rfsource</module>
		<module>../cse.rssm.gui.wizards.site</module>
		<module>../cse.rssm.gui.wizards.tuner</module>
		<module>../cse.rssm.gui.wizards.uplink</module>
		<module>../cse.rssm.gui.wizards.remoteunit</module>
		<module>../cse.rssm.gui.wizards.ptc</module>
		<module>../cse.rssm.gui.wizards.network</module>
		<module>../cse.rssm.gui.editors.activitymode</module>
		<module>../cse.rssm.gui.editors.network</module>
		<module>../cse.rssm.gui.editors.ptc</module>
		<module>../cse.rssm.gui.editors.remoteunit</module>
		<module>../cse.rssm.gui.editors.rfsource</module>
		<module>../cse.rssm.gui.editors.rfsourceinput</module>
		<module>../cse.rssm.gui.editors.site</module>
		<module>../cse.rssm.gui.editors.tuner</module>
		<module>../cse.rssm.gui.dialogs.export</module>
		<module>../cse.rssm.gui.dialogs.signaltest</module>
		<module>../cse.rssm.gui.views.alarming</module>
		<module>../cse.rssm.gui.views.alarming.tests</module>
		<module>../cse.rssm.gui.views.sites</module>
		<module>../cse.rssm.gui.views.networks</module>
		<module>../cse.rssm.gui.views.signalgrid</module>
		<module>../cse.rssm.gui.views.signalchart</module>
		<module>../cse.rssm.gui.views.signalchart.tests</module>
		<module>../cse.rssm.gui.perspectives.client</module>
		<module>../cse.rssm.gui.feature.cdo</module>
		<module>../cse.rssm.gui.feature.emf</module>
		<module>../cse.rssm.gui.feature.rcp</module>
		<module>../cse.rssm.gui.feature.nebula</module>
		<module>../cse.rssm.gui.feature</module>
		<module>../cse.rssm.gui</module>
		<module>../cse.rssm.gui.tests</module>
		<module>../repository</module>
	</modules>

	<build>
		<plugins>
			<plugin>
				<groupId>org.eclipse.tycho</groupId>
				<artifactId>tycho-maven-plugin</artifactId>
				<version>${tycho.version}</version>
				<extensions>true</extensions>
			</plugin>
			<plugin>
				<groupId>org.eclipse.tycho</groupId>
				<artifactId>tycho-compiler-plugin</artifactId>
				<version>${tycho.version}</version>
				<configuration>
					<compilerArgument>-warn:+discouraged,forbidden</compilerArgument>
				</configuration>
			</plugin>
			<plugin>
				<groupId>org.eclipse.tycho</groupId>
				<artifactId>tycho-surefire-plugin</artifactId>
				<version>${tycho.version}</version>
				<configuration>
					<argLine>-Xmx512m ${maven.test.jvmargs}</argLine>
					<failIfNoTests>false</failIfNoTests>
				</configuration>
			</plugin>
			<plugin>
				<groupId>org.eclipse.tycho</groupId>
				<artifactId>target-platform-configuration</artifactId>
				<version>${tycho.version}</version>
				<configuration>
				<executionEnvironment>JavaSE-17</executionEnvironment>
				     <target>
						<file>../cse.rssm.gui/cse.rssm.gui.target</file>
					</target>
					<environments>
						<environment>
							<os>win32</os>
							<ws>win32</ws>
							<arch>x86_64</arch>
						</environment>
					</environments>
				</configuration>
			</plugin>
			<plugin>
				<groupId>org.apache.maven.plugins</groupId>
				<artifactId>maven-surefire-plugin</artifactId>
				<version>2.7.1</version>
				<configuration>
					<forkMode>pertest</forkMode>
					<argLine>${maven.test.jvmargs}</argLine>
				</configuration>
			</plugin>
		</plugins>
	</build>
</project>
`
repo pom
`<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
	<modelVersion>4.0.0</modelVersion>
	<packaging>eclipse-repository</packaging>

	<parent>
		<groupId>cse.rssm.gui</groupId>
		<artifactId>cse.rssm.gui.parent</artifactId>
		<version>3.13.0-SNAPSHOT</version>
		<relativePath>../cse.rssm.gui.parent</relativePath>
	</parent>

	<artifactId>cse.rssm.gui.repository</artifactId>
		
	  <build>
    <plugins>
      <plugin>
        <groupId>org.eclipse.tycho</groupId>
        <artifactId>tycho-p2-director-plugin</artifactId>
        <version>2.7.4</version>
        <executions>
          <execution>
            
            <id>materialize-products</id>
            <goals>
              <goal>materialize-products</goal>
            </goals>
          </execution>
          <execution>
            
            <id>archive-products</id>
            <goals>
              <goal>archive-products</goal>
            </goals>
          </execution>
        </executions>
        
        <configuration>
          <products>
            <product>
              <id>develop</id>
              <rootFolder>application</rootFolder>
            </product>
          </products>
        </configuration>
      </plugin>
      </plugins>
      </build>
	
</project>`

**Comment by mickaelistria** [comment]
I'm not sure it's related, but maybe you should try Maven 3.8.5.

**Reply by jdebsucks** [reply]
Worth a shot but it didnt change anything

**Reply by jdebsucks** [reply]
I think I fixed it. The feature file contained
`<feature
      id="cse.rssm.gui.feature"
      label="Application Dependencies"
      version="3.13.0.qualifier"
      provider-name="Uplink Software Engineering"
      plugin="cse.rssm.gui"
      os="linux,macosx,win32"
      ws="cocoa,gtk,win32"
      nl="en_US"
      arch="x86,x86_64">
`
when I removed the os/ws/nl/arch lines it started working. is this a bug?
